hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/access/NonStrictAccessDelegate.java

```diff
@@ -35,17 +39,18 @@
 
 	private final InfinispanDataRegion region;
 	private final AdvancedCache cache;
-	private final AdvancedCache writeCache;
-	private final AdvancedCache putFromLoadCache;
+	private final FunctionalMap.ReadWriteMap<Object, Object> writeMap;
```

Comment:
~Shouldn't this be WriteOnlyMap? We're only writing, we don't care about previous value.~

Ah, these need to be RW so that the injectable component can check previous value? Does it apply to all uses?

Reply:
Yes, we always merge values, because order in which the commands are applied is not guaranteed. That's why we need the version from Hibernate.